src/kubernetes_api_objects/role.rs

```diff
@@ -0,0 +1,297 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+use crate::kubernetes_api_objects::api_resource::*;
+use crate::kubernetes_api_objects::common::*;
+use crate::kubernetes_api_objects::dynamic::*;
+use crate::kubernetes_api_objects::error::ParseDynamicObjectError;
+use crate::kubernetes_api_objects::marshal::*;
+use crate::kubernetes_api_objects::object_meta::*;
+use crate::kubernetes_api_objects::resource::*;
+use crate::pervasive_ext::string_map::StringMap;
+use crate::pervasive_ext::string_view::StringView;
+use vstd::prelude::*;
+use vstd::seq_lib::*;
+use vstd::vec::*;
+
+verus! {
+
+/// This definition is a wrapper of Role defined at
+/// https://github.com/Arnavion/k8s-openapi/blob/v0.17.0/src/v1_26/api/rbac/v1/role.rs.
+/// It is supposed to be used in exec controller code.
+///
+/// More detailed information: https://kubernetes.io/docs/reference/access-authn-authz/rbac/.
+
+#[verifier(external_body)]
+pub struct Role {
+    inner: deps_hack::k8s_openapi::api::rbac::v1::Role,
+}
+
+impl Role {
+    pub spec fn view(&self) -> RoleView;
+
+    #[verifier(external_body)]
+    pub fn default() -> (role: Role)
+        ensures
+            role@ == RoleView::default(),
+    {
+        Role {
+            inner: deps_hack::k8s_openapi::api::rbac::v1::Role::default(),
+        }
+    }
+
+    #[verifier(external_body)]
+    pub fn metadata(&self) -> (metadata: ObjectMeta)
+        ensures
+            metadata@ == self@.metadata,
+    {
+        todo!()
+    }
+
+
+    #[verifier(external_body)]
+    pub fn set_metadata(&mut self, metadata: ObjectMeta)
+        ensures
+            self@ == old(self)@.set_metadata(metadata@),
+    {
+        self.inner.metadata = metadata.into_kube();
+    }
+
+    #[verifier(external_body)]
+    pub fn set_policy_rules(&mut self, policy_rules: Vec<PolicyRule>)
+        ensures
+            self@ == old(self)@.set_policy_rules(policy_rules@.map_values(|policy_rule: PolicyRule| policy_rule@)),
+    {
+        self.inner.rules = std::option::Option::Some(
+            policy_rules.vec.into_iter().map(|p| p.into_kube()).collect()
+        )
+    }
+
+    #[verifier(external)]
+    pub fn into_kube(self) -> deps_hack::k8s_openapi::api::rbac::v1::Role {
+        self.inner
+    }
+
+    #[verifier(external_body)]
+    pub fn api_resource() -> (res: ApiResource)
+        ensures
+            res@.kind == Kind::CustomResourceKind,
```

Comment:
This should be `res@.kind == Kind::RoleKind`. There was a copy-paste bug made by me but it is fixed now.